infor-design/enterprise
# 2433 - Add Fullsize view support for Modals and Contextual Action Panels

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR adds a feature to the Modal and Contextual Action Panel components, allowing those components to take up the full size available on a screen with their contents.  The feature is "opt-in" and puts this functionality behind a setting called `fullsize`, which can either be false (default), "always", or "responsive" to only become full screen below the "phone-to-tablet" breakpoint.

Some additional work in this PR made the following changes:
- The `setTimeout()` code that wrapped some animation-related functions, and some event triggers, has been replaced with callbacks registered to the RenderLoop.
- Some settings in the Contextual Action Panel that were previously top-level settings have been deprecated in favor of a new `modalSettings` object, which contains all those same settings.  The Contexual Action Panel has been modified to pass that object directly as the underlying Modal's settings.  Deprecation warnings and backwards compatibility for the old settings have been added at the Contextual Action Panel level, and documentation has been updated accordingly.

**Related github/jira issue (required)**:
Closes #2433 

**Steps necessary to review your pull request (required)**:
Pull this branch, build, and run the app.

_To test the Modal portion of these changes:_
- Open http://localhost:4000/components/modal/example-fullsize-always.html
- Click the button to open the modal.  The modal should open and take up the full screen width and height.  The buttons at the bottom should be completely present and not be cut off.
- Open http://localhost:4000/components/modal/example-fullsize-responsive.html.  Make sure the screen size is above the "phone to tablet" breakpoint (larger than 767px wide)
- Click the button to open the modal.  The modal should appear normally (not full size)
- Resize the viewport to be less than 767px wide.  The modal should shift and take up the full screen width and height.

_To test the CAP:_
- Open http://localhost:4000/components/contextualactionpanel/test-fullsize-always.html
- Click the button to open the CAP.  The CAP should open and take up the full screen width and height.
- Open http://localhost:4000/components/contextualactionpanel/test-fullsize-responsive.html.  Make sure the screen size is above the "phone to tablet" breakpoint (larger than 767px wide)
- Click the button to open the CAP.  The CAP should appear normally (not full size)
- Resize the viewport to be less than 767px wide.  The CAP should shift and take up the full screen width and height.

_To test horizontal scrolling inside of a modal pane:_
- Open http://localhost:4000/components/contextualactionpanel/test-horizontal-scrolling.html
- Click the button to open the CAP.
- Try to scroll the area inside the CAP horizontally.  It should work with no issues.

**Included in this Pull Request**:
